tools/errorprone-checks/src/main/java/io/spine/tools/check/vbuilder/matcher/CustomProtobufType.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.check.vbuilder.matcher;
+
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.predicates.TypePredicate;
+import com.google.protobuf.Message;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Type;
+
+import static com.google.errorprone.matchers.method.MethodMatchers.instanceMethod;
+import static com.google.errorprone.matchers.method.MethodMatchers.staticMethod;
+import static com.google.errorprone.predicates.TypePredicates.isDescendantOf;
+
+/**
+ * A predicate which matches custom (i.e. non-Google) Protobuf types.
+ */
+final class CustomProtobufType implements TypePredicate {
+
+    private static final long serialVersionUID = 0L;
+
+    @SuppressWarnings("DuplicateStringLiteralInspection") // Used in another context.
+    private static final String GOOGLE_PROTOBUF_PACKAGE = "com.google.protobuf";
```

Comment:
We use a different approach is `IsNotGoogleProto`. I.e. we test for "google" to be included in the package name. That seems to be a more clever solution — see the numerous cases [here](https://github.com/googleapis/googleapis). 

It would also be good to extract the predicate from both scenario implementations and reuse.

Reply:
Let's discuss.